File: solutions/Remove-Duplicates-from-Sorted-List/Remove-Duplicates-from-Sorted-List.cpp

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        if(!head)
        {
            return head;
        }
        ListNode *p=head->next;
        ListNode *q=head;
        while(p)
        {
            if(p->val != q->val)
            {
                q->next=p;
                q=p;
                
            }
            p=p->next;
        }
        q->next=NULL;
        return head;
    }
};


 

File: solutions/Remove-Duplicates-from-Sorted-List/Remove-Duplicates-from-Sorted-List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function (head) {
  if (head === null) {
    return head;
  }
  let slow = head;
  let fast = head.next;
  while (fast !== null) {
    if (fast.val !== slow.val) {
      slow.next = fast;
      slow = fast;
    }
    fast = fast.next;
  }
  slow.next = null;
  return head;
};